### PR TITLE
feat: implementation of docker compose replica for gst dlstreamer

### PR DESCRIPTION
**Pull request by jim-wang-intel**
## PR Checklist

- [x] Added label to the Pull Request for easier discoverability and search
- [x] Commit Message meets guidelines as indicated in the URL https://github.com/intel-retail/retail-use-cases/blob/main/CONTRIBUTING.md
- [x] Every commit is a single defect fix and does not mix feature addition or changes
- [ ] Unit Tests have been added for new changes
- [x] Updated Documentation as relevant to the changes
- [ ] All commented code has been removed
- [ ] If you've added a dependency, you've ensured license is compatible with repository license and clearly outlined the added dependency.
- [ ] PR change contains code related to security
- [ ] PR introduces changes that breaks compatibility with other modules (If YES, please provide details below)

## What are you changing?

details in https://github.com/intel-retail/automated-self-checkout/issues/507

## Issue this PR will close

close: **https://github.com/intel-retail/automated-self-checkout/issues/507**

## Test Instructions if applicable

- git clone this PR
- if first time, you need to get the content of submodules via running `make update-submodules`

### build dlstreamer Docker image

```bash
(cd use-cases/dlstreamer; make build)
```

### example run commands

- run one gst dlstreamer pipeline and use the default object detection only (yolov5s.sh)

    ```bash
    make run-gst
    ```

- run three gst dlstreamer pipelines and use the default object detection only (yolov5s.sh)

    ```bash
    PIPELINE_COUNT=3 make run-gst
    ```

- run two gst dlstreamer pipelines and use the object detection with classification (yolov5s_effnetb0.sh)

    ```bash
    PIPELINE_SCRIPT=yolov5s_effnetb0.sh PIPELINE_COUNT=2 make run-gst
    ```

- shutdown Docker containers

    ```bash
    make down-gst
    ```

- clean up the output results

    ```bash
    make clean-results
    ```

### CODE REVIEW

File: use-cases/dlstreamer/docker-compose_replicas_gst.yml [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+      replicas: ${PIPELINE_COUNT}
+    network_mode: "host"
+    entrypoint: /script/entrypoint.sh --pipeline_script_choice ${PIPELINE_SCRIPT}
+    env_file:
+      - ./res/gst.env
+      - ./res/yolov5-cpu.env
+    environment:
+      - CONTAINER_NAME="gst0"
+      - GRPC_PORT=9001
+      - INPUTSRC=rtsp://localhost:8554/camera_1
```
Comment (brian-intel) [inline use-cases/dlstreamer/docker-compose_replicas_gst.yml:114]: Now that camera simulator containers are static we can launch only the vehicle-bike and remove the others.

File: use-cases/dlstreamer/docker-compose_replicas_gst.yml [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        -f rtsp
+        -rtsp_transport
+        tcp
+        rtsp://localhost:8554/camera_2
+        "
+    depends_on:
+      - camera-simulator
+    volumes:
+      - ../../performance-tools/sample-media:/home/pipeline-server/sample-media
+  # OvmsServer:
```
Comment (brian-intel) [inline use-cases/dlstreamer/docker-compose_replicas_gst.yml:79]: Remove commented service

File: use-cases/dlstreamer/docker-compose_replicas_gst.yml [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+      - INPUTSRC=rtsp://localhost:8554/camera_1
+      - RENDER_MODE=0 #RENDER_MODE=1 will work only after running xhost +local:docker
+      - DISPLAY=$DISPLAY
+      - PIPELINE_SCRIPT=${PIPELINE_SCRIPT}
+    volumes:
+      - ../../results:/tmp/results
+      - ~/.Xauthority:/home/dlstreamer/.Xauthority
+      - /tmp/.X11-unix
+      - ~/.cl-cache:/home/pipeline-server/.cl-cache
+      - ../../performance-tools/sample-media/:/home/pipeline-server/vids
```
Comment (brian-intel) [inline use-cases/dlstreamer/docker-compose_replicas_gst.yml:123]: Sample-media won't be directly used and can be removed

File: use-cases/dlstreamer/docker-compose_replicas_gst.yml [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+      - ./res/:/home/pipeline-server/envs
+      - ./extensions:/home/pipeline-server/extensions
+      - ./framework-pipelines:/home/pipeline-server/framework-pipelines
```
Comment (brian-intel) [inline use-cases/dlstreamer/docker-compose_replicas_gst.yml:126]: extensions and framework-pipelines are copied in the Dockerfile. The volumes can be removed.

**Review by antoniomtz — APPROVED** [review]
LGTM. All commands  ran successfully

**Review by brian-intel — APPROVED** [review]
LGTM

**Pull request merged**